js/src/carousel.js

```diff
@@ -181,6 +181,16 @@ class Carousel {
     }
 
     if (this._config && this._config.interval && !this._isPaused) {
+      this._activeElement = SelectorEngine.findOne(SELECTOR_ACTIVE_ITEM, this._element)
+
+      if (this._activeElement) {
+        const activeElementInterval = parseInt(this._activeElement.getAttribute('data-interval'), 10)
+
+        if (activeElementInterval) {
+          this._config.interval = activeElementInterval
+        }
+      }
```

Comment:
I think we should create a separate private method to do that because earlier we do : 

```js
      const nextElementInterval = parseInt(nextElement.getAttribute('data-interval'), 10)
      if (nextElementInterval) {
        this._config.defaultInterval = this._config.defaultInterval || this._config.interval
        this._config.interval = nextElementInterval
      } else {
        this._config.interval = this._config.defaultInterval || this._config.interval
      }
```

here : https://github.com/mitchellbryson/bootstrap/blob/carousel-data-bug/js/src/carousel.js#L476-L482

So it would be better if we can do something like that : 

```js
this._updateInterval(this._activeElement);
```

which will update correctly the interval

Reply:
Thanks. Is it worth patching V4 as well?